packages/teams-js/src/public/search.ts

```diff
@@ -0,0 +1,131 @@
+import { sendMessageToParent } from '../internal/communication';
+import { registerHandler, removeHandler } from '../internal/handlers';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { runtime } from './runtime';
+
+/**
+ * Allows you to interact with the host search experience
+ *
+ * @beta
+ */
+export namespace search {
+  const onChangeHandlerName = 'search.queryChange';
+  const onClosedHandlerName = 'search.queryClose';
+  const onExecutedHandlerName = 'search.queryExecute';
+
+  /**
+   * This interface contains information pertaining to the search term in the host search experience
+   */
+  export interface SearchQuery {
+    /** The current search term in the host search experience */
+    searchTerm: string;
+
+    // TODO: may need some sort of timestamp or sequence value to
```

Comment:
You should remove TODO comments and also come to a decision on anything called out as a TODO. In this case, I had a guess that you would want some sort of timestamp about when the search was executed to help developers ensure they were responding to search terms in the correct order. You should either do that, or remove the commented out code entirely. (Although if you *don't* do this I'd want to hear about how developers can feel confident they're getting events in the correct order)

Reply:
Are messages between the hubs and the apps not guaranteed to be delivered in the order they're dispatched?

Reply:
Question from jobutle:

"Are messages between the hosts and the apps not guaranteed to be delivered in the order they're dispatched?"

Reply:
To the best of my knowledge, we don't make any guarantee that messaages are received in order by the app-sdk. We're at the mercy of JS async code and the post messaging layer. The messages *might* arrive in order, but I can't promise you that. 

Whether or not our messages are passed in order, in my experience getting events from things like search boxes that are linked to rapid typing (on key press) are sometimes not delivered in order by the lower level code handling those things. 

I'd recommend that you offer developers some way of making sure they're processing those events in the correct order